unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/HUDFactory.cs

```diff
@@ -71,7 +71,7 @@ public virtual async UniTask<IHUD> CreateHUD(HUDElementID hudElementId, Cancella
                 return new TermsOfServiceHUDController();
             case HUDElementID.FRIENDS:
                 return new FriendsHUDController(DataStore.i,
-                    FriendsController.i,
+                    Environment.i.serviceLocator.Get<IFriendsController>(),
```

Comment:
`serviceLocator.Get` is somewhat expensive due to having to do a dictionary lookup. Cache the instance and reuse it.

Reply:
But this is called only once per hud element, meaning it won't be called more than a few times (less than 5) in the startup, what would be the point of optimizing that?
Get in a dictionary is an O(1) operation

Reply:
It piles up quickly because even though it's O(1), it has overhead.

For example look at this frame I picked in the editor.
![image](https://user-images.githubusercontent.com/50231608/235598140-fc88d5d9-171c-46e4-aa3d-8fb6903b1bd1.png)

We have 16.6ms to render a frame at 60fps, spending 0.40ms in getting instances from a Service Locator taking 2.5% of frame time seems unnecessary if it can be avoided.

Obviously I picked up the worst case in the sample frames I took and you are right, in this case we are talking about initialization of a system, not a hotpath recurrently called every frame. But tidy it up here will help avoiding the pattern in the future by mimicking.

Reply:
Okay, in the future I'll prevent calling the get, but I think the problem in that picture is that it called 92 times in a single frame, while only one call takes nothing